src/tools/cephfs_mirror/Mirror.cc

```diff
@@ -556,7 +556,9 @@ void Mirror::update_fs_mirrors() {
   {
     std::scoped_lock locker(m_lock);
     for (auto &[filesystem, mirror_action] : m_mirror_actions) {
-      auto failed_restart = mirror_action.fs_mirror && mirror_action.fs_mirror->is_failed() &&
+      bool failed = (mirror_action.fs_mirror && mirror_action.fs_mirror->get_init_finished())?
```

Comment:
Have you tested this out by running one tests for like 50 odd times in teuthology?

Reply:
Which test?

Reply:
There are no test failures in /a/yuriw-2024-03-01_20:51:20-fs-squid-distro-default-smithi/7578112. But the unwinding finds a coredump and the run failed:

```
 72841 2024-03-02T01:16:56.577 DEBUG:teuthology.run_tasks:Unwinding manager internal.sudo
 72842 2024-03-02T01:16:56.588 INFO:teuthology.task.internal:Restoring /etc/sudoers...
 72843 2024-03-02T01:16:56.588 DEBUG:teuthology.orchestra.run.smithi134:> sudo mv -f /etc/sudoers.orig.teuthology /etc/sudoers
 72844 2024-03-02T01:16:56.616 DEBUG:teuthology.run_tasks:Unwinding manager internal.coredump
 72845 2024-03-02T01:16:56.626 DEBUG:teuthology.orchestra.run.smithi134:> sudo sysctl -w kernel.core_pattern=core && sudo bash -c 'fo\
       r f in `find /home/ubuntu/cephtest/archive/coredump -type f`; do file $f | grep -q systemd-sysusers && rm $f || true ; done' &\
       & rmdir --ignore-fail-on-non-empty -- /home/ubuntu/cephtest/archive/coredump
 72846 2024-03-02T01:16:56.706 INFO:teuthology.orchestra.run.smithi134.stdout:kernel.core_pattern = core
 72847 2024-03-02T01:16:56.740 DEBUG:teuthology.orchestra.run.smithi134:> test -e /home/ubuntu/cephtest/archive/coredump
 72848 2024-03-02T01:16:56.754 WARNING:teuthology.task.internal:Found coredumps on ubuntu@smithi134.front.sepia.ceph.com, flagging ru\
       n as failed
```

Reply:
This test run -  /a/yuriw-2024-03-01_20:51:20-fs-squid-distro-default-smithi/7578112, yes. Just that there should not be any crashes in the job and therefore the tests should pass.

Reply:
@vshankar That job started 30 tests. So it ran all the tests in test_mirroring.py (30 tests).

[jcollin@telecast 7578112]$ grep 'Starting test:' teuthology.log | wc -l
30

Do you mean run the entire test_mirroring.py 50 times?

Reply:
The tests are passed: https://pulpito.ceph.com/jcollin-2024-03-29_15:07:50-fs:mirror-wip-jcollin-testing_29032024-distro-default-smithi/